### PR TITLE
Remove eval and toSource usage

### PR DESCRIPTION
**Pull request by kewisch**
Ideally I can enable CSP in calendar app to improve security when I eventually want to enable HTML descriptions I have other blocking issues in calendar itself that prevents me from using the full advantages of CSP but this eval is a blocker long term.

### CODE REVIEW

File: build/ical.js [hunk 1]
```diff
@@ -53,6 +53,39 @@ ICAL.helpers = {
     return obj;
   },
 
+  isArray: function(o) {
+    return o && (o instanceof Array || typeof o == "array");
+  },
+
+  clone: function(aSrc, aDeep) {
```
Comment (lightsofapollo) [inline build/ical.js:60]: I would write a  basic unit test for this function, you may have noticed I am somewhat of a fanatic =/

Comment (kewisch) [inline build/ical.js:60]: Which is a good thing :-) I'll do so!

**Comment by lightsofapollo** [comment]
I would write a test for clone though

@kewisch r=lightsofapollo

**Comment by lightsofapollo** [comment]
landing working to hook up recurring in gaia now.

**Pull request merged**